controllers/storagecluster/provider_server.go

```diff
@@ -434,3 +444,62 @@ func RandomString(l int) string {
 
 	return string(bytes)
 }
+
+func getOnboardingJobObject(instance *ocsv1.StorageCluster) *batchv1.Job {
+
+	return &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      onboardingJobName,
+			Namespace: instance.Namespace,
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					RestartPolicy:      corev1.RestartPolicyOnFailure,
+					ServiceAccountName: onboardingJobName,
+					Containers: []corev1.Container{
+						{
+							Name:    onboardingJobName,
+							Image:   os.Getenv(onboardingSecretGeneratorImage),
+							Command: []string{"/usr/local/bin/onboarding-secret-generator"},
+							Env: []corev1.EnvVar{
+								{
+									Name:  "OPERATOR_NAMESPACE",
+									Value: instance.Namespace,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (o *ocsProviderServer) createJob(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
+	if os.Getenv(onboardingSecretGeneratorImage) == "" {
```

Comment:
1. we need to establish a watch on the secret in `SetupWithManager` and need to create a job only when the public key isn't found

Reply:
Okay i will check this out.